public/scripts/upload.js

```diff
@@ -731,6 +723,20 @@ function main() {
     },
   );
 
+  // observe fileListViewModel: imageUploads[].length
+  // side effect: generate preview mockup
+  mobx.reaction(
+    () => viewModel.fileList.imageUploads.length,
+    (newLen) => {
+      if (viewModel.fileList.imageUploads.length !== newLen) {
+        console.error("unexpected mobx error, image upload length not matched");
+        return;
+      }
+      const newImage = viewModel.fileList.imageUploads[newLen - 1];
+      viewModel.generatePreviewMockup(newImage);
+    },
+  );
+
```

Comment:
it will trigger generate preview for last image when removing uploaded images from, that's why I add `viewModel.generatePreviewMockup(newImage);` in add function, maybe add check `if newImage.previewURL != null`?

Reply:
Lemme check it with 

```
if newLen > oldLen
```

> maybe add check if newImage.previewURL != null?

I think this is workaround. This side effect should only be triggered by target observable. Adding flags to side effect is, IMO, anti-pattern

Reply:
Updated, thanks yayun!